hack/drone.jsonnet

```diff
@@ -0,0 +1,323 @@
+  name: std.format("%s", name),
+  image: build_container,
+  commands: [if target == "" then std.format("make %s", name) else std.format("make %s", target)],
+  environment: build_env_vars+environment,
+  volumes: step_volumes,
+  depends_on: [x.name for x in depends_on],
+};
+
+
+local pipeline(name, steps=[]) = {
+  kind: "pipeline",
+  name: name,
+  node: node,
+  services: if name != "notify" then [docker] else [],
+  volumes: if name != "notify" then pipeline_volumes else [],
+  clone: {
+    disable: true,
+  },
+  steps: if name != "notify" then [clone, buildkit] + steps else steps,
+};
+
+// Apps
+
+local machined = step("machined", [clone]);
+local osd = step("osd", [clone]);
+local trustd = step("trustd", [clone]);
+local proxyd = step("proxyd", [clone]);
+local ntpd = step("ntpd", [clone]);
+local osctl_linux = step("osctl-linux", [clone]);
+local osctl_darwin = step("osctl-darwin", [clone]);
+
+// Artifacts
+
+local rootfs =  step("rootfs", [machined, osd, trustd, proxyd, ntpd]);
+local initramfs = step("initramfs", [rootfs]);
+local installer = step("installer", [rootfs]);
+local container = step("container", [rootfs]);
+local image_azure = step("image-azure", [installer]);
+local image_gce = step("image-gce", [installer]);
+local kernel = step("kernel", [clone]);
+local iso = step("iso", [installer]);
+
+
+// CAPI.
+
+local push_image_azure = step("push-image-azure", [image_azure], creds_env_vars);
+local push_image_gce = step("push-image-gce", [image_gce], creds_env_vars);
+
+// Tests
+
+local lint = step("lint", [rootfs]);
+local unit_tests = step("unit-tests", [rootfs]);
+local basic_integration = step("basic-integration", [container, osctl_linux]);
+local capi = step("capi", [basic_integration], creds_env_vars);
+local e2e_integration_azure = step("e2e-integration-azure", [capi, push_image_azure], {PLATFORM: "azure"}, "e2e-integration");
+local e2e_integration_gce = step("e2e-integration-gce", [capi, push_image_gce], {PLATFORM: "gce"}, "e2e-integration");
+local conformance_azure = step("conformance-azure", [capi, push_image_azure], {PLATFORM: "azure", CONFORMANCE: "run"}, "e2e-integration");
+local conformance_gce = step("conformance-gce", [capi, push_image_gce], {PLATFORM: "gce", CONFORMANCE: "run"}, "e2e-integration");
+
+// Steps
+
+local apps = [
+    machined,
+    osd,
+    trustd,
+    proxyd,
+    ntpd,
+    osctl_linux,
+    osctl_darwin,
+];
+
+local artifacts = [
+  rootfs,
+  initramfs,
+  installer,
+  container,
+];
+
+local tests = [
+  lint,
+  unit_tests,
+  basic_integration,
+];
+
+local e2e_setup = apps + artifacts + [
+  image_azure,
+  image_gce,
+  push_image_azure,
+  push_image_gce,
+  basic_integration,
+  capi,
+];
+
+local e2e = e2e_setup + [
+  e2e_integration_azure,
+  e2e_integration_gce,
+];
+
+local conformance = e2e_setup + [
+  conformance_azure,
+  conformance_gce,
+];
+
+
+local release_step ={
+  name: "release",
+  image: "plugins/github-release",
+  settings:{
```

Comment:
did you want to try to include the `note: ` bit to automatically pull in the ami names?

Reply:
Shimmying that in to the way the `step` function works made things super ugly. In the `step` function I'm hoping to force us to use a single command for _most_ things. I found that it makes the steps a lot more clear and helps align the `Makefile` and `drone.jsonnet` file.

Reply:
Plus with everything we talked about today in regard to publish and AWS asset instead of the AMIs, this felt less important.